src/script.cpp

```diff
@@ -1500,11 +1500,27 @@ bool is_stealth_info_type(const operation_stack& ops)
     return ops.size() == 2 &&
         ops[0].code == opcode::return_ &&
         ops[1].code == opcode::special &&
-        ops[1].data.size() >= hash_size;
+        ops[1].data.size() <= 40;
```

Comment:
It's sometimes hazardous to rely on documentation. The [default maximum OP_RETURN in the satoshi client](https://github.com/bitcoin/bitcoin/pull/5286) is 80 bytes.

The documented [stealth transaction format](https://wiki.unsystem.net/en/index.php/DarkWallet/Stealth) indicates that 38 bytes is required:

```
RETURN <version:1=6> <nonce:4> <P:33>
```

However this documentation is also dated.

The version byte was dropped, the nonce was moved to the right of the public key, the public key was restricted to 32 bytes (sign fixed by convention), and the nonce length was made variable (any number of bytes up to 80 at this point). This was done to allow the value to match that of more arbitrary data in the blockchain by not being fixed. It also served to increase nonce space. I believe this change was documented but I can't locate it presently.

So the proper check for length of a stealth `OP_RETURN is >= hash_size` (32 bytes). The `read_ephemkey()` function assumes this symbol, so it's both a safety and readability issue.